src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/Cas1SpaceBookingController.kt

```diff
@@ -161,12 +163,28 @@ class Cas1SpaceBookingController(
   ): ResponseEntity<Unit> {
     userAccessService.ensureCurrentUserHasPermission(UserPermission.CAS1_SPACE_BOOKING_RECORD_ARRIVAL)
 
+    val arrivalDateAndTime = if (cas1NewArrival.arrivalDateTime != null) {
+      Pair(
+        cas1NewArrival.arrivalDateTime!!.toLocalDate(),
+        cas1NewArrival.arrivalDateTime!!.toLocalDateTime().toLocalTime(),
+      )
+    } else {
+      val arrivalDate = cas1NewArrival.arrivalDate
+        ?: throw BadRequestProblem(invalidParams = mapOf("arrivalDate" to "is required"))
+      val arrivalTime = cas1NewArrival.arrivalTime
+        ?: throw BadRequestProblem(invalidParams = mapOf("arrivalTime" to "is required"))
+      Pair(
+        arrivalDate,
+        LocalTime.parse(arrivalTime),
+      )
+    }
```

Comment:
is this new block worth of a private function? 
The jury's out on this tbf as this function is not that big even with this - occurred to me as an option :)

Reply:
Yep I agree, probably would be neater but as this is temporary code until we remove cas1NewArrival.arrivalDateTime i think probably fine as is